### PR TITLE
Fixes the antag panel not being removed on mind loss 

### PR DESCRIPTION
**Pull request by SuperNovaa41**
## About The Pull Request
Since action buttons are stored on the mob, they must be manually removed. These are removed on body transfer, however the proc for that is not *always* called, (i.e. mind swap), so this ensures that it always does by removing it anytime the mind has been transfered.

## Why It's Good For The Game
these panels should be removed...

Tested on surgery and mind swap spell, both work

## Changelog

:cl:
fix: Antag panels will no longer remain on bodies after the mind has been taken out.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/_common/antag_datum.dm [hunk 1]
```diff
@@ -65,10 +65,13 @@ GLOBAL_LIST_EMPTY(antagonists)
 	GLOB.antagonists += src
 	typecache_datum_blacklist = typecacheof(typecache_datum_blacklist)
 
+	RegisterSignal(owner, COMSIG_MIND_TRANSFERRED, .proc/on_body_removal)
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/_common/antag_datum.dm:68]: Shouldn’t this be on on_gain? Or apply_innate_whatever

Comment (SuperNovaa41) [reply code/modules/antagonists/_common/antag_datum.dm]: Yes it should

File: code/modules/antagonists/ashwalker/ashwalker.dm [hunk 2]
```diff
@@ -23,9 +23,12 @@
 /datum/antagonist/ashwalker/get_team()
 	return ashie_team
 
-/datum/antagonist/ashwalker/on_body_transfer(mob/living/old_body, mob/living/new_body)
+/datum/antagonist/ashwalker/on_body_removal(mob/living/old_body)
 	. = ..()
 	UnregisterSignal(old_body, COMSIG_MOB_EXAMINATE)
```
Comment (MrMelbert) [inline code/modules/antagonists/ashwalker/ashwalker.dm:28]: These args are incorrect

`/datum/antagonist/proc/on_body_removal(datum/mind/mind, mob/living/old_body)`

`/datum/antagonist/ashwalker/on_body_removal(mob/living/old_body)`

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/ashwalker/ashwalker.dm (hunk 2 above)
Comment (SuperNovaa41) [reply code/modules/antagonists/ashwalker/ashwalker.dm:28]: oh right, forgot to update that, thanks

**Comment by SuperNovaa41** [comment]
I'll come back to this later

**Pull request closed without merging**